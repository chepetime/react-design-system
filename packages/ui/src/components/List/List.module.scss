@use "./../../styles" as styles;

:root {
  --list-background-color: #{styles.$pink-500};
}

:global .dark {
  --list-background-color: #{styles.$indigo-500};
}

.List {
  @include styles.styles-reset;
  @include styles.text-wrapping;

  background: var(--list-background-color);

  font-size: 1rem;

  @include styles.respond-to("md") {
    font-size: 1.5rem;
  }
  @include styles.respond-to("lg") {
    font-size: 2rem;
  }
  @include styles.feature-to("hover") {
    &:hover {
      background-color: pink;
    }
  }
}
