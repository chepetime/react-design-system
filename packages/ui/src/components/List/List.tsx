import * as React from "react";
import classnames from "classnames";
import css from "./List.module.scss";

export interface ListProps {
  children?: React.ReactNode;
}

export const List = React.forwardRef<HTMLDivElement, ListProps>(
  (props: ListProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.List)} ref={ref}>
        <p>List component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

List.displayName = "List";
