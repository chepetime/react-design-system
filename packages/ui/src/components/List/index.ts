export { List } from "./List";
export type { ListProps } from "./List";