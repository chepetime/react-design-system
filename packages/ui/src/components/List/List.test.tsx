import React from "react";
import { render } from "@testing-library/react";
import { List } from "./List";

describe("List", () => {
  test("renders the List component", () => {
    render(<List />);
  });
});
