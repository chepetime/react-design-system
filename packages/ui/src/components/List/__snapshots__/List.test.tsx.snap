// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List matches snapshot 1`] = `
<div>
  <div
    class="List"
  >
    <p>
      List component working!
    </p>
    <div />
  </div>
</div>
`;
